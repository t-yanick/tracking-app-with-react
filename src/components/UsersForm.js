import React, { useState } from 'react';
import PropTypes from 'prop-types';

const UsersForm = ({ handleSubmit, btnName }) => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  const onSubmit = (e) => {
    e.preventDefault();
    handleSubmit(username, password);
  };

  return (
    <div>
      <form onSubmit={onSubmit} className="form">
        <div className="form__group">
          <input
            type="text"
            name="username"
            placeholder="Enter your username"
            value={username}
            onChange={(e) => setUsername(e.target.value)}
          />
        </div>
        <div className="form__group">
          <input
            type="password"
            name="password"
            placeholder="Enter your password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
        </div>
        <button type="submit" className="btn mb3">{btnName}</button>
      </form>
    </div>
  );
};

UsersForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  btnName: PropTypes.string,
};

UsersForm.defaultProps = {
  btnName: 'Save',
};

export default UsersForm;