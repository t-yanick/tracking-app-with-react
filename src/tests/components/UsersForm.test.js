import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import UsersForm from '../../components/UsersForm';
import createTestStore from '../fixtures/createTestStore';

describe('UsersForm', () => {
  let store;
  beforeEach(() => {
    store = createTestStore();
  });
  test('should match with snapshot', () => {
    const handleSubmit = jest.fn();
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <UsersForm handleSubmit={handleSubmit} />
          </MemoryRouter>
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});