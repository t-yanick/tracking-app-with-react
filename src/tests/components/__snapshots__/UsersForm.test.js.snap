// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersForm should match with snapshot 1`] = `
<div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="form__group"
    >
      <input
        name="username"
        onChange={[Function]}
        placeholder="Enter your username"
        type="text"
        value=""
      />
    </div>
    <div
      className="form__group"
    >
      <input
        name="password"
        onChange={[Function]}
        placeholder="Enter your password"
        type="password"
        value=""
      />
    </div>
    <button
      className="btn mb3"
      type="submit"
    >
      Save
    </button>
  </form>
</div>
`;
